packages/vue/src/composables/form.ts

```diff
@@ -88,14 +88,24 @@ export function useForm<
 		})
 	}
 
+	function resetSubmissionState() {
+		successful.value = failed.value = recentlyFailed.value = recentlySuccessful.value = false
+		clearTimeout(timeoutIds.recentlySuccessful!)
+		clearTimeout(timeoutIds.recentlyFailed!)
+		progress.value = undefined
+	}
+
 	/**
 	 * Resets the form to its initial values.
 	 */
 	function reset(...keys: P[]) {
```

Comment:
I disagree with the changes to `reset`. This function should reset `fields` and that's it. I'm open to discuss having another function to reset everything though.